Sprint 1

MikhailSamigullin opened the pull request.

MikhailSamigullin committed:
Create file structure and markup

Markup html files and css. Use sass prerpos with scss syntax.
Pages are without popups.

MikhailSamigullin committed:
Install Parsel and Handlebars. Update HTML

Just try use Parsel and Handlebars together.

MikhailSamigullin committed:
Hot fix for deploy on Netlify

MikhailSamigullin committed:
Try move folder dist on deploy branch

MikhailSamigullin committed:
Delete dist directory

MikhailSamigullin committed:
Delete .parcel-cache directory

MikhailSamigullin committed:
Delete .gitignore

MikhailSamigullin committed:
Delete package-lock.json

MikhailSamigullin committed:
Delete package.json

MikhailSamigullin committed:
Delete .parcelrc

MikhailSamigullin committed:
Make necessary changes to submit the project

All .html are .hbs files now.
Create main partial parts.
Change css structure: create single .css file and fix double styles.
Delete comments and empty spaces.

MikhailSamigullin committed:
Update README.md

MikhailSamigullin committed:
FIX previous commit

MikhailSamigullin committed:
Change type on commonjs

MikhailSamigullin committed:
FIX again package.json

MikhailSamigullin committed:
FIX

MikhailSamigullin committed:
FIX

MikhailSamigullin committed:
FIX start script

gennady-bars commented on src/style/_variables.scss:
```diff
@@ -0,0 +1,54 @@
+// Second colors
+
+$black: #000000;
+$grey: #999999;
```
* Отлично, что цвета вынесены в `css`-константы

gennady-bars commented on app.js:
```diff
@@ -0,0 +1,9 @@
+const express = require('express');
+const app = express();
+const port = 3000;
```
**Можно лучше**

Лучше дать возможность поменять порт из консоли, если понадобится: `const PORT = process.env.PORT || 3000;`  . Если не задан порт, тогда будет `3000` по умолчанию.

MikhailSamigullin replied:
Поменял. Если запустить $env:PORT=5000, то изменяет порт на 5000

gennady-bars commented on package.json:
```diff
@@ -0,0 +1,31 @@
+  "keywords": [],
+  "author": "MikhailSamigullin",
+  "license": "ISC",
+  "devDependencies": {
+    "@anikin/parcel-transformer-handlebars": "^1.0.9",
+    "parcel": "^2.7.0",
+    "express": "^4.18.2"
```
у Вас в `devDependencies` такие же пакеты, что и в `dependencies`. Этого не должно быть. Либо там, либо там нужно расположить их

gennady-bars replied:
В `devDependencies` указывается то, что необходимо для сборки и разработки приложения, а в `dependencies` нужно указывать то, что необходимо непосредственно для работы приложения (express, uuid). Все линтеры, тесты и тд нужно только в `devDependencies` указывать, чтобы зря не раздувать кол-во пакетов в собранном для деплоя приложении.

gennady-bars commented on package.json:
```diff
@@ -0,0 +1,31 @@
+  "dependencies": {
+    "@anikin/parcel-transformer-handlebars": "^1.0.9",
+    "parcel": "^2.7.0",
+    "express": "^4.18.2"
```
Лучше задавать **фиксированные версии** зависимостям. Если нечаянно удалить `package-lock.json` и попытаться повторно установить зависимости с помощью команды `npm install`, то **npm** установит последние версии зависимостей, что может привести к ошибкам.

gennady-bars replied:
лучше все зависимости фиксировать

gennady-bars commented on package.json:
```diff
@@ -0,0 +1,31 @@
+    "express": "^4.18.2"
+  },
+  "engines": {
+    "node": ">=16.13.2",
```
* Отлично, что ограничиваете версию `node`

gennady-bars commented on src/pages/index.hbs:
```diff
@@ -0,0 +1,29 @@
+                <br> Чат.
+            </h1>
+            {{>button link="./login/login.hbs" cls="enter__button" label="Авторизоваться"}}   
+            <div class="enter__link-list">
```
Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту

gennady-bars commented on src/pages/index.hbs:
```diff
@@ -0,0 +1,29 @@
+            </h1>
+            {{>button link="./login/login.hbs" cls="enter__button" label="Авторизоваться"}}   
+            <div class="enter__link-list">
+                <a href="#" class="enter__link">Контакты</a>
```
Список ссылок выглядит как список, значит, и отрисовываться должен с помощью `ul` и `li`. Это стандартная практика, чтобы улучшить `UX` (это нужно для парсеров сайтов и для спец возможностей браузера). A внутри `li` уже тег `a` будет

gennady-bars replied:
**Можно лучше**

Чтобы сделать сайт более доступным для пользователя, нужно стараться придерживаться **семантической верстки**.
Несколько полезных статей на эту тему:
•	[Что такое семантическая вёрстка и зачем она нужна](https://htmlacademy.ru/blog/html/semantics)
•	[Семантика для циников](https://www.youtube.com/watch?v=B5pomLr059U)

gennady-bars commented on src/pages/index.hbs:
```diff
@@ -0,0 +1,29 @@
+    <title>Просто. Чат.</title>
+</head>
+<body>
+    <div class="index_container">
```
По семантике нужно помещать весь главный контент страницы в тег `main`.

MikhailSamigullin replied:
Изменил первый div на main на всех страницах

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


* Верстка страниц очень достойная
* Отлично работают переходы между страницами по ссылкам
* `Readme` отлично оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Хорошая структура папок и файлов
* Отлично, что цвета вынесены в `css`-константы

но есть некоторые недочеты: 

* По семантике нужно помещать весь главный контент страницы в тег `main`.
* лучше добавить иконку сайта  `fav.icon`, чтобы не было ошибки в консоли https://disk.yandex.ru/i/YEzgH1Z9Q0R5AQ
* у Вас в `devDependencies` такие же пакеты, что и в `dependencies`. Этого не должно быть. Либо там, либо там нужно расположить их по одному экземпляру
*  Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту
* Список ссылок выглядит как список, значит, и отрисовываться должен с помощью `ul` и `li`. Это стандартная практика, чтобы улучшить `UX` (это нужно для парсеров сайтов и для спец возможностей браузера). A внутри `li` уже тег `a` будет
* По семантике нужно помещать весь главный контент страницы в тег `main`.


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

MikhailSamigullin commented on src/pages/index.hbs:
```diff
@@ -0,0 +1,29 @@
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
```
Подключил favicon через partials. Работает, но не знаю насколько это правильно. Наверное, лучший вариант, чем хардкодить на каждой странице.

MikhailSamigullin committed:
FIX after review

Some changes:
 - import favicon,
 - use right tags in HTML(more semantic),
 - use environment in app.js
 - hard versions in package.json

MikhailSamigullin committed:
Update README.md

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.


* кнопка сабмита должна быть внутри тега `form`. В следующем спринте об этом не забудьте, пожалуйста
* **Обработчик сабмита** нужно навешивать только на тег `form` с событием `submit`, а не на кнопку сабмита с событием `click`, так как сабмит формы происходит ещё при нажатии `Enter`, и он не будет работать, если навесить обработчик клика на кнопку только. **Это нужно исправить везде, где есть инпуты и форма**
* В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`
